libs/wingsdk/src/shared/sandbox.ts

```diff
@@ -12,8 +14,138 @@ export interface SandboxOptions {
   readonly log?: (internal: boolean, level: string, message: string) => void;
 }
 
+type ProcessRequest = {
+  fn: string;
+  args: any[];
+};
+
+type ProcessResponse =
+  | {
+      type: "resolve";
+      value: any;
+    }
+  | {
+      type: "reject";
+      reason: Error;
+    };
+
 export class Sandbox {
-  private loaded = false; // "true" after first run (module is loaded into context)
+  private createBundlePromise: Promise<void>;
+  private entrypoint: string;
+  private readonly timeouts: NodeJS.Timeout[] = [];
+  private readonly options: SandboxOptions;
+
+  constructor(entrypoint: string, options: SandboxOptions = {}) {
+    this.entrypoint = entrypoint;
+    this.options = options;
+    this.createBundlePromise = this.createBundle();
+  }
+
+  public async cleanup() {
+    for (const timeout of this.timeouts) {
+      clearTimeout(timeout);
+    }
+  }
+
+  private async createBundle() {
+    const workdir = await mkdtemp(path.join(tmpdir(), "wing-bundles-"));
+
+    // wrap contents with a shim that handles the communication with the parent process
+    // we insert this shim before bundling to ensure source maps will work correctly
+    let contents = (await readFile(this.entrypoint)).toString();
+
+    // log a warning if contents includes __dirname or __filename
+    if (contents.includes("__dirname") || contents.includes("__filename")) {
```

Comment:
this will only catch refs to `__dirname` and `__filename` at the root module, no required modules.

Reply:
I'm aware (but don't know of any other solutions)